### PR TITLE
Create efficient service for retrieving testflight program

### PR DESCRIPTION
**Pull request by adamjcampbell**
# 📝 Summary of Changes

Changes proposed in this pull request:

- Create `getTestFlightProgram` service function

# 🧐🗒 Reviewer Notes

## 💁 Example

Functionality for `sync pull` remains unchanges

## 🔨 How To Test
Test that `sync pull` remains unchanged

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -31,3 +31,4 @@ Temporary Items
 /*.xcodeproj
 xcuserdata/
 config/auth.yml
+config/apps
```
Comment (DechengMa) [inline .gitignore:34]: finally we have this in .ignore file, 🎉

File: Sources/AppStoreConnectCLI/Commands/TestFlight/Sync/TestFlightPullCommand.swift [hunk 2]
```diff
@@ -28,13 +28,8 @@ struct TestFlightPullCommand: CommonParsableCommand {
     func run() throws {
         let service = try makeService()
 
-        // TODO: A new service function should be created to efficiently gather these models
-        let apps = try service.listApps(bundleIds: filterBundleIds)
-        let identifiers = apps.map { app in AppLookupIdentifier.appId(app.id) }
-        let testers = try service.listBetaTesters(filterIdentifiers: identifiers, limit: 200)
-        let groups = try service.listBetaGroups(filterIdentifiers: identifiers)
+        let testflightProgram = try service.getTestFlightProgram()
```
Comment (DechengMa) [inline Sources/AppStoreConnectCLI/Commands/TestFlight/Sync/TestFlightPullCommand.swift:31]: I can see `filterBundleIds` is no longer get used, we might want to remove this 
```
//    @Option(
//        parsing: .upToNextOption,
//        help: "Filter by only including apps with the specified bundleIds in the configuration"
//    ) var filterBundleIds: [String]
```
Or are we going to use this in the future?

**Review by DechengMa — APPROVED** [review]
Looks good

File: Sources/AppStoreConnectCLI/Commands/TestFlight/Sync/TestFlightPullCommand.swift (hunk 2 above)
Comment (adamjcampbell) [reply Sources/AppStoreConnectCLI/Commands/TestFlight/Sync/TestFlightPullCommand.swift:31]: That was a mistake! Thanks for catching it 👍

**Pull request merged**